### PR TITLE
Implement @range on short

### PR DESCRIPTION
**Pull request by LukeMathWalker**
## Motivation and Context
See #1968 

## Checklist

- [x] I have updated `CHANGELOG.next.toml` if I made changes to the smithy-rs codegen or runtime crates

----

_By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice._

### CODE REVIEW

File: codegen-core/common-test-models/constraints.smithy [hunk 1]
```diff
@@ -193,6 +200,9 @@ structure ConstrainedHttpBoundShapesOperationInputOutput {
     @httpHeader("X-Range-Integer")
     rangeIntegerHeader: RangeInteger,
 
+    @httpHeader("X-Range-Short")
+    rangeShortHeader: RangeShort,
```
Comment (hlbarber) [inline codegen-core/common-test-models/constraints.smithy:204]: The other HTTP bindings which are compatible with short are [@httpLabel](https://smithy.io/2.0/spec/http-bindings.html#httplabel-trait), and [@httpResponseCode](https://smithy.io/2.0/spec/http-bindings.html#httpresponsecode-trait). Do we want to also add coverage for them?

File: codegen-core/common-test-models/constraints.smithy [hunk 2]
```diff
@@ -497,6 +543,18 @@ integer MaxRangeInteger
 @range(min: 69, max: 69)
 integer FixedValueInteger
 
+@range(min: -0, max: 10)
+integer RangeShort
```
Comment (hlbarber) [inline codegen-core/common-test-models/constraints.smithy:547]: Am I misunderstanding, isn't this (and below) supposed to be `short RangeShort`?

https://smithy.io/2.0/spec/simple-types.html#short

File: codegen-core/common-test-models/constraints.smithy (hunk 1 above)
Comment (LukeMathWalker) [reply codegen-core/common-test-models/constraints.smithy:204]: Yes, that's [here](https://github.com/awslabs/smithy-rs/pull/2034/files#diff-d1ac0e3b7b50015c41afa00ec53ce786a3e1744483220a0afab4c1c1a0d7503fR187)

Comment (hlbarber) [reply codegen-core/common-test-models/constraints.smithy:204]: Turns out I'm blind

File: codegen-core/common-test-models/constraints.smithy (hunk 2 above)
Comment (LukeMathWalker) [reply codegen-core/common-test-models/constraints.smithy]: Ups, yes - fixed in https://github.com/awslabs/smithy-rs/pull/2034/commits/3fd6655869effd811e2c242c40a07d9db11a9932

**Review by hlbarber — APPROVED** [review]
LGTM

**Pull request merged**